tests/test_qasm_examples.py

```diff
@@ -1,17 +1,136 @@
+    p = re.compile(r"(if)\s*\(\s*(\w+)\s*(==|!=|>=|<=|<|>)\s*(\w+)\s*\)")
+    match = p.match(line)
+    # if an if-string is found, "build up" a correctly formatted string
+    if match:
+        ifstring = match.group(1) + " (" + match.group(2) + " " + match.group(3) + " " + match.group(4) + ")"
+        return ifstring + line[match.span()[1]:]
+    return line
+
+
+def fix_spaces(line):
+    """Helper function to format spaces before/after paranthesis and after commas correctly"""
+    line = ",".join(re.split(r"\s*,\s*", line))
+    line = "(".join(re.split(r"\s*\(\s*", line))
+    line = ") ".join(re.split(r"\s*\)\s*", line))
+    return clean(line)
+
+
+# for variable names that are used by Python, and cannot be used as SymPy symbols
+invalid_var_names = ["lambda"]
+def fix_math_expr(line):
+    """Helper function to format math expressions correctly"""
+
+    # remove the if-parts from the string (doens't support maths in if-statments)
+    if "if" in line:
+        line = re.split(r"if\s*\(.+?\)\s*", line)[-1]
+
+    p = re.compile(r"\(.+\)")
+    res = p.search(line)
+    invalid_var_names_list = []
+    if res and "=" not in res.group() and res.group() not in ["False", "True"]:
+        p_match = res.group()
+        free_params = re.findall(r"[a-zA-Z]\w*", p_match)
+
+        for i, p in enumerate(free_params):
+            if p in invalid_var_names:
+                free_params[i] = p + "_var"
+                p_match = p_match.replace(p, free_params[i])
+                invalid_var_names_list.append((p, free_params[i]))
+
+        sympy.var(free_params)
+
+        p_replace = p_match
+        if p_replace[0] == "(":
+            p_replace = p_replace[1:-1]
+        p_replace = ",".join([str(eval(r)) for r in p_replace.split(",")])
+        for var in invalid_var_names_list:
+            p_replace = p_replace.replace(var[1], var[0])
+            p_match = p_match.replace(var[1], var[0])
+        line = line.replace(p_match, "(" + p_replace + ")")
+
+    return line
+
+
 def clean(qasm_line):
     """Helper function to strip out comments, whitespace, and empty braces from a line from a qasm file."""
     line = qasm_line.split("//")[0]  # comments
-    line = " ".join(re.split(r"[{]\s*[}]", line))  # empty braces ( )
-    line = " ".join(re.split(r"[(]\s*[)]", line))  # empty braces { }
     line = " ".join(line.split())  # normalize whitespace
     return line
 
+
+def format_code(qasm_lines):
+    """Format the code"""
+    formatted_qasm_lines = [clean(line) for line in qasm_lines if clean(line)]
+    formatted_qasm_lines = split_at_semicol(formatted_qasm_lines)
+    formatted_qasm_lines = [fix_spaces(line) for line in formatted_qasm_lines]
+    formatted_qasm_lines = [fix_ifs(line) for line in formatted_qasm_lines]
+    formatted_qasm_lines = fix_braces(formatted_qasm_lines)
+    formatted_qasm_lines = [fix_math_expr(line) for line in formatted_qasm_lines]
+
+    return formatted_qasm_lines
```

Comment:
Despite how scary some of the above helper functions look, it's nice to see things here so clearly.